### PR TITLE
Remove `sessionConfig` from constructor

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

`sessionConfig` in the end has been totally removed from the constructor of the library. It didn't make sense to have so many different forms of `sessionConfig`, so there is now only one place to specify this, in the context.

One side effect has been on the OGM, where I have added a top-level `database` config option, which passes the value through to the context when querying. This has been added to the migration guide.

## Complexity

Complexity: Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-3655.surge.sh

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3655.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 63 | 56 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 115 | 124 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 70 | 92 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 4913 | 4373 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 4315 | 4591 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 63 | 78 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 67 | 83 | 2174 |
|  aggregations.DeeplyNestedCount | 13070327 | 13070333 | 6929 | 7051 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 222 | 251 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 132 | 199 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 246 | 277 | 3003 |
|  connections.Connection | 14082 | 14082 | 92 | 96 | 2174 |
|  connections.NestedConnection | 45468 | 45472 | 176 | 181 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 105 | 119 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 326 | 528 | 1040 |
|  create.SimpleMutation | 7 | 7 | 96 | 125 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 845 | 895 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153617 | 153617 | 254 | 339 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 449 | 427 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 252 | 326 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 68 | 92 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 160 | 197 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 52 | 70 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 75 | 86 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 30 | 35 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 76 | 81 | 2174 |
|  query.QueryWhere | 9724 | 9696 | 51 | 61 | 2169 |
|  query.SimpleQueryWithNestedWhere | 9902 | 9874 | 76 | 88 | 2169 |
|  query.Nested | 10096041 | 10096041 | 13668 | 14108 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 13996 | 13268 | 2004000 |
|  query.OrFilterOnRelationships | 43483 | 41931 | 274 | 368 | 2126 |
|  query.OrFilterOnRelationshipsAndNested | 37831 | 34992 | 330 | 341 | 2126 |
|  query.QueryWithNestedIn | 13593 | 13175 | 78 | 86 | 1281 |
|  query.NestedConnectionWhere | 9834 | 9834 | 89 | 90 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9911 | 10033 | 130 | 140 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19086 | 19218 | 207 | 252 | 1580 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 300 | 339 | 1134 |
|  query.Fulltext | 80 | 80 | 48 | 48 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 79 | 70 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 152 | 128 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1414 | 1449 | 376 | 510 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 88 | 80 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 153 | 151 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 257 | 375 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 92 | 95 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 97 | 191 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 412 | 609 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 378 | 483 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 179 | 227 | 2002 |

</details>
        
Old Schema Generation: 1:07.111 (m:ss.mmm)
Schema Generation: 1:09.600 (m:ss.mmm)
Old Subgraph Schema Generation: 1:27.304 (m:ss.mmm)
Subgraph Schema Generation: 1:27.772 (m:ss.mmm)

File: packages/graphql/tests/e2e/deprecated/delete-relationship-via-delete-additional-labels.e2e.test.ts [hunk 1]
```diff
@@ -104,7 +104,13 @@ describe("Delete Subscriptions when only nodes are targeted - when nodes employ
                 subscriptions: new TestSubscriptionsPlugin(),
             },
         });
-        server = new ApolloTestServer(neoSchema);
+        // eslint-disable-next-line @typescript-eslint/require-await
```
Comment (angrykoala) [inline packages/graphql/tests/e2e/deprecated/delete-relationship-via-delete-additional-labels.e2e.test.ts:107]: Why is this disable required?

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/tests/e2e/deprecated/delete-relationship-via-delete-additional-labels.e2e.test.ts (hunk 1 above)
Comment (darrellwarde) [reply packages/graphql/tests/e2e/deprecated/delete-relationship-via-delete-additional-labels.e2e.test.ts:107]: The Apollo context function is typed to always be async... But the contents don't always await anything. So ESLint complains. 🙈

**Pull request merged**